plugins/common/socket/socket.go

```diff
@@ -4,18 +4,20 @@ import (
 	"bufio"
 	"crypto/tls"
 	"fmt"
+	"github.com/shirou/gopsutil/v3/cpu"
 	"io"
 	"net"
 	"net/url"
 	"regexp"
 	"strings"
+	"time"
 
 	"github.com/influxdata/telegraf"
 	"github.com/influxdata/telegraf/config"
 	tlsint "github.com/influxdata/telegraf/plugins/common/tls"
 )
 
-type CallbackData func(net.Addr, []byte)
+type CallbackData func(net.Addr, []byte, time.Time)
```

Comment:
Is this really required?

Reply:
Yes, because the parsing is started in a separate goroutine (pool), it might start several milliseconds after it was received.  Not passing the receive time to the processing callback and taking the time when processing is started results in a very inconsistent timestamping.

See my description and measurements in #15884